internal/controllers/networking/loadbalancer_controller_test.go

```diff
@@ -104,4 +108,78 @@ var _ = Describe("LoadBalancerReconciler", func() {
 			g.Expect(loadBalancerRouting.NetworkRef.UID).To(BeEquivalentTo(network.UID))
 		}).Should(Succeed())
 	})
+
+	It("should allocate internal IPs for internal load balancers", func() {
+		By("creating a network")
+		network := &networkingv1alpha1.Network{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    ns.Name,
+				GenerateName: "network-",
+			},
+		}
+		Expect(k8sClient.Create(ctx, network)).To(Succeed())
+
+		By("creating a prefix")
+		rootPrefix := &ipamv1alpha1.Prefix{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    ns.Name,
+				GenerateName: "lb-",
+			},
+			Spec: ipamv1alpha1.PrefixSpec{
+				IPFamily: corev1.IPv4Protocol,
+				Prefix:   commonv1alpha1.MustParseNewIPPrefix("10.0.0.0/24"),
+			},
+		}
+		Expect(k8sClient.Create(ctx, rootPrefix)).To(Succeed())
+
+		By("creating an internal load balancer")
+		loadBalancer := &networkingv1alpha1.LoadBalancer{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    ns.Name,
+				GenerateName: "internal-lb-",
+			},
+			Spec: networkingv1alpha1.LoadBalancerSpec{
+				Type:       networkingv1alpha1.LoadBalancerTypeInternal,
+				NetworkRef: corev1.LocalObjectReference{Name: network.Name},
+				IPFamilies: []corev1.IPFamily{corev1.IPv4Protocol},
+				IPs: []networkingv1alpha1.IPSource{
+					{
+						Ephemeral: &networkingv1alpha1.EphemeralPrefixSource{
+							PrefixTemplate: &ipamv1alpha1.PrefixTemplateSpec{
+								Spec: ipamv1alpha1.PrefixSpec{
+									IPFamily:  corev1.IPv4Protocol,
+									ParentRef: &corev1.LocalObjectReference{Name: rootPrefix.Name},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, loadBalancer)).To(Succeed())
+
+		By("waiting for the prefix to be created with the correct spec and become ready")
+		prefix := &ipamv1alpha1.Prefix{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: loadBalancer.Namespace,
+				Name:      networkingv1alpha1.LoadBalancerIPIPAMPrefixName(loadBalancer.Name, 0),
+			},
+		}
+		Eventually(Object(prefix)).Should(SatisfyAll(
+			HaveField("Spec.IPFamily", corev1.IPv4Protocol),
+			HaveField("Spec.ParentRef", &corev1.LocalObjectReference{Name: rootPrefix.Name}),
+			HaveField("Spec.PrefixLength", int32(32)),
+			HaveField("Spec.Prefix", commonv1alpha1.MustParseNewIPPrefix("10.0.0.0/32")),
+			HaveField("Status.Phase", ipamv1alpha1.PrefixPhaseAllocated),
+		))
+
+		By("asserting it get's an internal IP")
+		Eventually(Object(loadBalancer)).Should(HaveField("Status.IPs", ContainElements(*commonv1alpha1.MustParseNewIP("10.0.0.0"))))
+
+		By("deleting the load balancer")
+		Expect(k8sClient.Delete(ctx, loadBalancer)).To(Succeed())
+
+		By("waiting for it to be gone")
+		Eventually(Get(loadBalancer)).Should(Satisfy(apierrors.IsNotFound))
```

Comment:
The deletion part is not required / can be removed.

Reply:
Done